pkg/utils/utils.go

```diff
@@ -284,6 +284,41 @@ func CleanCachedNetConf(cRefPath string) error {
 	return nil
 }
 
+// SetVFEffectiveMAC will try to set the mac address on a specific VF interface
+// the function will also validate that the mac address was configured as expect
+// it will return an error if it didn't manage to configure the vf mac address
+// or the mac is not equal to the expect one
+// retries 20 times and wait 100 milliseconds
+func SetVFEffectiveMAC(netLinkManager NetlinkManager, netDeviceName string, macAddress string) error {
+	hwaddr, err := net.ParseMAC(macAddress)
+	if err != nil {
+		return fmt.Errorf("failed to parse MAC address %s: %v", macAddress, err)
+	}
+
+	linkObj, err := netLinkManager.LinkByName(netDeviceName)
+	if err != nil {
+		return err
+	}
+
+	err = Retry(20, 100*time.Millisecond, func() error {
+		if err := netLinkManager.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
+			return err
+		}
+
+		linkObj, err = netLinkManager.LinkByName(netDeviceName)
```

Comment:
I'd create a local copy for linkObj since this could fail. In the next retry, linkObj will be nil (passed to linkSetHardwareAddrr which might panic)

Reply:
great catch!

Reply:
do we really need this (LinkByName) here ? we already getting linkObj in L198